OpenSim/Common/ComponentSocket.h

```diff
@@ -374,6 +387,21 @@ class Socket : public AbstractSocket {
         connectInternal(*objT);
     }
 
+    /** Find a component using a partial path or component name of the correct
+     * type anywhere in the model (using Component::findComponent()) and
+     * connect to that component.
+     *
+     * Throws an exception If you provide only a component name and the
+     * model has multiple components with that nume.
+     * */
+    void findAndConnect(const ComponentPath& connectee) override {
+        const auto* comp =
+            getOwner().getRoot().template findComponent<T>(connectee);
+        OPENSIM_THROW_IF(!comp, Exception,
```

Comment:
Should we introduce a `ComponentNotFound` exception?

Reply:
There's a `ComponentNotFoundOnSpecifiedPath` here: https://github.com/opensim-org/opensim-core/blob/defd589d5a3824ddf212943c3851d6ea6187b1ba/OpenSim/Common/Component.h#L94-L110

Reply:
Done.